src/containers/BookList.js

```diff
@@ -1,28 +1,51 @@
 import React from "react";
 import { connect } from "react-redux";
-import PropTypes from "prop-types";
+import { removeBook } from "../actions/index";
 import Book from "../components/Book";
 
-const BookList = (props) => {
-  const { books } = props;
-  return (
-    <table>
-      <tr>
-        <th>Book ID</th>
-        <th>Title</th>
-        <th>Category</th>
-      </tr>
-      {books.map((book) => (<Book key={book.id} book={book} />))}
-    </table>
-  );
-};
+class BookList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleRemoveBook = this.handleRemoveBook.bind(this);
+  }
 
-BookList.propTypes = {
-  books: PropTypes.string.isRequired,
-};
+  handleRemoveBook(book) {
+    const { removeBook } = this.props;
+    removeBook(book);
+  }
+
+  render() {
+    const { bookList } = this.props;
+
+    return (
+      <table>
+        <thead>
+          <tr>
+            <th>Book ID</th>
+            <th>Title</th>
+            <th>Category</th>
+          </tr>
+        </thead>
+        <tbody>
+          {bookList.map((book) => (
+            <Book
+              key={book.id}
+              book={book}
+              clickHandler={this.handleRemoveBook}
+            />
+          ))}
+        </tbody>
+      </table>
+    );
+  }
+}
+
```

Comment:
Please we need to validate our props by doing the `Typechecking with PropTypes`, that was the mean reason you added it in milestone 1;
![image](https://user-images.githubusercontent.com/31897434/88877145-c1c43500-d21c-11ea-8c1b-1b6452874de5.png)